core/src/test/java/org/testcontainers/junit/DockerComposeContainerVolumeRemovalTest.java

```diff
@@ -0,0 +1,96 @@
+package org.testcontainers.junit;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.rnorth.ducttape.unreliables.Unreliables;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.containers.DockerComposeContainer;
+
+import java.io.File;
+import java.util.LinkedHashSet;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Parameterized.class)
+public class DockerComposeContainerVolumeRemovalTest {
+
+    public DockerComposeContainerVolumeRemovalTest(
+        final boolean removeVolumes,
+        final boolean shouldVolumesBePresentAfterRunning
+    ) {
+        this.removeVolumes = removeVolumes;
+        this.shouldVolumesBePresentAfterRunning = shouldVolumesBePresentAfterRunning;
+    }
+
+    public final boolean removeVolumes;
+
+    public final boolean shouldVolumesBePresentAfterRunning;
+
+    @Parameterized.Parameters(name = "removeVolumes = {0}")
+    public static Object[][] params() {
+        return new Object[][] { { true, false }, { false, true } };
+    }
+
+    @Test
+    public void performTest() {
+        final File composeFile = new File("src/test/resources/compose-test.yml");
+
+        final AtomicReference<String> volumeName = new AtomicReference<>("");
+        try (
+            DockerComposeContainer environment = new DockerComposeContainer<>(composeFile)
+                .withExposedService("redis", 6379)
+                .withRemoveVolumes(removeVolumes)
+                .withRemoveImages(DockerComposeContainer.RemoveImages.ALL)
+        ) {
+            environment.start();
+
+            volumeName.set(volumeNameForRunningContainer("_redis_1"));
+            final boolean isVolumePresentWhileRunning = isVolumePresent(volumeName.get());
+            assertThat(isVolumePresentWhileRunning).as("the container volume is present while running").isEqualTo(true);
+        }
+
+        Unreliables.retryUntilSuccess(
```

Comment:
Just for further reference, let's use [Awaitility](http://www.awaitility.org/). I'll do the changes before merge it.

Reply:
I'll keep that in mind for the future. Thank you!